dev-docs/11-how-wp-app-initialize.md

```diff
@@ -0,0 +1,50 @@
+## Initialization of the WP App
+
+The `Enpii_Base_Bootstrap` class is designed to initialize and set up the Enpii Base framework within a WordPress environment. It includes methods to manage both CLI and web mode operations, ensuring the proper configuration of WordPress application hooks, redirects, and actions.
+
+The main goal of the initialization process is to prepare the necessary folder structure and manage database setup or migration tasks for the plugin. The `initialize` method is the entry point for this process, triggered early in the `enpii-base-init.php` file. It checks whether the WordPress content directory is loaded and then performs the appropriate setup actions based on whether the environment is in CLI or web mode.
+
+This will create a laravel application `wp_app()` (a DI container https://code.tutsplus.com/tutorials/digging-in-to-laravels-ioc-container--cms-22167) contains everything we need.
+
+### 1. Command-Line Interface (CLI) Mode Setup
+
+In CLI mode, the setup process is managed through two specific command-line instructions:
+
+```bash
+wp enpii-base prepare       # Set up the correct folder structure
+wp enpii-base wp-app:setup  # Run migrations and copy necessary assets
+```
+- `enpii-base prepare`: Prepares and sets up the required folder structure for the plugin.
+- `enpii-base wp-app:setup`: Runs the necessary migrations and copies the required assets to their appropriate locations.
+
+If the environment is running in CLI mode, the `Enpii_Base_Helper::is_console_mode()` method confirms it. The CLI initialization action is then registered via `static::register_cli_init_action()`, binding it to the cli_init hook with `Enpii_Base_Helper::wp_cli_init()` as the callback.
+
+### 2. Web Mode Setup
+
+In web mode, the plugin performs the following tasks:
+
+- **Prepare the necessary folder structure**: The Enpii Base plugin requires a Laravel-style folder structure to function correctly.
```

Comment:
We use a flexible folder structure, just the cache and storage folder, you can see the list of folders we create in the src